controlpanel/api/aws.py

```diff
@@ -357,8 +354,8 @@ def grant_bucket_access(role_name, bucket_arn, access_level, path_arns=[]):
     policy.put()
 
 
-def revoke_bucket_access(role_name, bucket_arn=None, path_arns=[]):
-    if not bucket_arn and not path_arns:
+def revoke_bucket_access(role_name, bucket_arn=None):
```

Comment:
I didn't test it or checked properly but from quick look of it you may be able to make the `bucket_arn` argument mandatory without breaking compatibility with the client code.

That's simpler and less sneaky and it would allow you to also get rid of the `if not bucket_arn` and the warning for something that shouldn't really happen unless there is a bug (from what I can understand at least)

Reply:
I think this probably should be the case, though I'm not certain enough so I think it'd be safest to leave it as is. Perhaps, the `bucket_arn` could not exist if a bucket has been deleted in S3 but not in the control panel (?)

Reply:
To answer your question the ARN should be generated using the AWS account ID and the bucket name so even when not present it shouldn't be blanck.

But of course it may not be passed to the function somehow - we can always change this later probably good call to be on the safe side 👍